### ISSUE TITLE
Support Chart.js@^2.4.0

### ISSUE THREAD

**Issue by sedouard** [vmpowerio/chartjs-node#7]
Seems like when this package is used we get the error message:

`ReferenceError: HTMLCanvasElement is not defined.`

Which possibly suggests that the node bindings for `canvas` are somehow to working.

**Comment by kdekooter** [comment]
Would this help? https://github.com/chartjs/Chart.js/issues/3696#issuecomment-266360502

**Comment by FilipeDominguesGit** [comment]
@kdekooter after reading that comment on chart.js early today i made this changes to jsdom Promise :
```
.then(window => {
            // these are probably not defined but just to be safe store them
            const canvas = require('canvas');
            const canvasMethods = ['HTMLCanvasElement'];

            Object.keys(window).forEach(property => {
            if (typeof global[property] === 'undefined') {
                global[property] = window[property];
            }
            });

            canvasMethods.forEach(method =>
            global[method] = window[method]
            );

            global['CanvasRenderingContext2D'] = canvas.Context2d;

            global.navigator = {
            userAgent: 'node.js'
            };


            const Chartjs = require('chart.js');
            if(configuration.options.plugins) {
                Chartjs.pluginService.register(configuration.options.plugins);
            }
           
            this._disableDynamicChartjsSettings(configuration);
            this._canvas = BbPromise.promisifyAll(window.document.getElementById('myChart'));
            this._ctx = this._canvas.getContext('2d');
           
            this._chart = new Chartjs(this._ctx, configuration);
            return this._chart;
        });
    }
```

And its working fine now with chart.js@2.4.0  :smile: .

**Comment by kdekooter** [comment]
@FilipeDominguesGit it was that simple huh ;-). Going to submit a PR shortly.

**Comment by sedouard** [comment]
I love waking up to PRs :). Thanks @kdekooter!

**Comment by sedouard** [comment]
Thanks guys, you rock! Published the a version chartjs-node@1.2.0 with these changes.
